RecoLocalCalo/HGCalRecProducers/plugins/HGCalLayerClusterProducer.cc

```diff
@@ -147,25 +159,38 @@ void HGCalLayerClusterProducer::produce(edm::Event& evt,
 
   algo->getEventSetup(es);
 
+  //make a map detid-rechit
+  // NB for the moment just host EE and FH hits
+  // timing in digi for BH not implemented for now
+  std::unordered_map<uint32_t, float> hitmap;
+
   switch(algoId){
   case reco::CaloCluster::hgcal_em:
     evt.getByToken(hits_ee_token,ee_hits);
     algo->populate(*ee_hits);
+    for(auto const& it: *ee_hits) hitmap[HGCalDetId(it.detid())] = it.time();
```

Comment:
I do not think the conversions to `HGCalDetId` are necessary or desirable. Just use `it.detid().rawId()` to get the raw value.